opal/class/opal_interval_tree.c

```diff
@@ -355,31 +356,54 @@ int opal_interval_tree_insert (opal_interval_tree_t *tree, void *value, uint64_t
     return OPAL_SUCCESS;
 }
 
+static int opal_interval_tree_compare_node(opal_interval_tree_node_t *node, uint64_t low, uint64_t high, void *data) {
+    if ((data && node->low == low && node->high == high && node->data == data) ||
+        (!data && node->low <= low && node->high >= high)) {
+        return 0;
+    }
+    if (node->low > low) {
+        return -1;
+    }
+    if (node->low < low) {
+        return 1;
+    }
+    if (node->high < high) {
+        return -1;
+    }
+    if (node->high > high) {
+        return 1;
+    }
+    if (node->data > data) {
```

Comment:
Is this `node->data > data` here intentional?  There are three reasons this stands out at first glace (without fully groking the code):
1. It seems strange to be comparing a `void * data` pointer with `>` operator.
2. This is new logic that did not exist before this PR.
3. There is no equivalent `<` test.

Perhaps you meant to use the `!=` operator?

Reply:
It is intended as a tie breaker. The intent is to allow nodes with identical ranges as long as the data differs. This covers the vader usage.  I can add a comment to that effect.